### PR TITLE
Symbolab: Add support for queries of the form "integral of x^2"

### PR DESCRIPTION
**Pull request by levaly**
https://duck.co/ia/view/symbolab

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by levaly** [comment]
https://duck.co/ia/view/symbolab

File: lib/DDG/Spice/Symbolab.pm [hunk 1]
```diff
@@ -7,9 +7,10 @@ spice is_cached => 1;
 spice to => 'https://www.symbolab.com/ddg?query=$1';
 spice wrap_jsonp_callback => 1;
 
-triggers startend => "calculate", "solve", "compute";
-handle remainder => sub {
-    if ($_ =~ /[\^\*\=\+\-\/\\]/ or $_ =~ /\b(sin|cos|tan|cot|csc|sec|ln|log|sqrt|integral)/i) {
+triggers startend => "calculate", "solve", "compute", "integral", "integrate", "integration", "antiderivative", "derivative", "derive", "differentiate", "derivation";
+handle query_lc => sub {
+    print "Recevied: $_\n";
```
Comment (GuiltyDolphin) [inline lib/DDG/Spice/Symbolab.pm]: Remove the debugging :wink:

**Comment by levaly** [comment]
@moollaza @GuiltyDolphin is there anything else to be done on this feature?

**Comment by GuiltyDolphin** [comment]
@levaly Looks pretty good to me. @jdorweiler Can we deploy this to beta?

**Comment by jdorweiler** [comment]
@levaly @GuiltyDolphin it's on beta now.

**Comment by GuiltyDolphin** [comment]
@levaly Thanks! :dolphin:

The result is incorrect for `integral of x^2` (gives `(x^4)/4 + C`), but I believe you are handling this elsewhere?

**Comment by levaly** [comment]
@GuiltyDolphin what is the url for beta?
I don't get any symbolab result for: https://beta.duckduckgo.com/?q=integral%20of%20x%5E2%20dx&ia=answer

**Comment by jdorweiler** [comment]
https://beta.duckduckgo.com/?q=integral+of+x%5E2+dx&ia=calculator

**Comment by levaly** [comment]
the problem with "integral of x^2 dx" is resolved.

File: lib/DDG/Spice/Symbolab.pm [hunk 2]
```diff
@@ -7,9 +7,9 @@ spice is_cached => 1;
 spice to => 'https://www.symbolab.com/ddg?query=$1';
 spice wrap_jsonp_callback => 1;
 
-triggers startend => "calculate", "solve", "compute";
-handle remainder => sub {
-    if ($_ =~ /[\^\*\=\+\-\/\\]/ or $_ =~ /\b(sin|cos|tan|cot|csc|sec|ln|log|sqrt|integral)/i) {
+triggers startend => "calculate", "solve", "compute", "integral", "integrate", "integration", "antiderivative", "derivative", "derive", "differentiate", "derivation";
+handle query_lc => sub {
+    if ($_ =~ /[\^\*\=\+\-\/\\]/ or $_ =~ /\b(sin|cos|tan|cot|csc|sec|ln|log|sqrt|integr|deriv|diff)/i) {
```
Comment (moollaza) [inline lib/DDG/Spice/Symbolab.pm]: @levaly there is a problem here: your regex is matching against the query for `integr` and the trigger is `integration`.

This means every query that contains the word "integration", "integrated", or ""integrate", etc seems like a valid query and triggers this IA

We need to fix this and likely add a second check for the presence of numbers `\d+` in the query in certain cases

Comment (moollaza) [inline lib/DDG/Spice/Symbolab.pm]: @levaly actually, you probably just need to add `\b` to the end of the 2nd regex unless you expect a specific character to follow those keywords?

Comment (levaly) [inline lib/DDG/Spice/Symbolab.pm]: I will make the change asap. Sorry for the late reply.

Comment (moollaza) [inline lib/DDG/Spice/Symbolab.pm]: > Sorry for the late reply.

No problem, and thanks a lot!

Comment (levaly) [inline lib/DDG/Spice/Symbolab.pm]: I created a PR with the fixes, and updated the unit tests accordingly.

**Pull request merged**